ARE/components/processor.averager/src/main/java/eu/asterics/component/processor/averager/AveragerInstance.java

```diff
@@ -177,6 +220,12 @@ public synchronized Object setRuntimePropertyValue(String propertyName, Object n
      * samples are summed but not divided
      */
     private synchronized void process(final double in) {
+        if (!propEnabled && buffer.size() == propBufferSize) {
```

Comment:
Why do you need the check ```... && buffer.size() == propBufferSize```

Reply:
the idea was somehow to wait until the buffer is full, but it does not make much sense as it is now. I'll remove it.